test/sol6/sanityRatesGasPrice.js

```diff
@@ -0,0 +1,328 @@
+const SanityRates = artifacts.require('SanityRatesGasPrice.sol');
+const MockReserveSanity = artifacts.require('MockReserveSanity.sol');
+const TestToken = artifacts.require('Token.sol');
+const {expectEvent, expectRevert} = require('@openzeppelin/test-helpers');
+const Helper = require('../helper.js');
+const {BPS, precisionUnits, ethAddress, zeroAddress} = require('../helper.js');
+const {web3} = require('@openzeppelin/test-helpers/src/setup');
+const BN = web3.utils.BN;
+const MAX_RATE = new BN(10).pow(new BN(25)); // 10**25
+
+let sanityRates;
+let admin;
+let operator;
+let numTokens = 5;
+let token;
+let tokens = [];
+let tokenDecimals = [];
+let rates = [];
+let buyRate = [];
+let sellRate = [];
+let reasonableDiffs = [];
+let gasPrice = web3.utils.toWei('100', 'gwei');
+
+contract('SanityRatesGasPrice', function (accounts) {
+  before('init globals.', async function () {
+    admin = accounts[0];
+    operator = accounts[5];
+
+    //init tokens
+    for (let i = 0; i < numTokens; i++) {
+      tokenDecimals[i] = new BN(15).add(new BN(i));
+      token = await TestToken.new(`test${i}`, 'tst' + i, tokenDecimals[i]);
+      tokens[i] = token.address;
+    }
+  });
+
+  describe('test sanity rates with maxGasPrice setting', function () {
+    before('init sanity rates.', async function () {
+      for (let i = 0; i < numTokens; i++) {
+        rates[i] = new BN(i + 1).mul(precisionUnits.div(new BN(10)));
+        reasonableDiffs[i] = new BN(i * 100);
+      }
+  
+      sanityRates = await SanityRates.new(admin, gasPrice);
+      await sanityRates.addOperator(operator);
+  
+      await sanityRates.setReasonableDiff(tokens, reasonableDiffs);
+    });
+
+    it('should test return rate 0 when rate has not been set yet.', async function () {
+      const rate0 = await sanityRates.getSanityRate(tokens[0], ethAddress);
+      Helper.assertEqual(rate0, 0, '0 rate expected');
+    });
+
+    it('should test setting sanity rates.', async function () {
+      await sanityRates.setSanityRates(tokens, rates, {from: operator});
```

Comment:
not this is never called by non operator address

Reply:
Added test.